src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -196,6 +259,12 @@ protected int doWork() {
             // read reference sequence one by one and write its metadata
             for (ReferenceSequence refSeq = refSeqFile.nextSequence(); refSeq != null; refSeq = refSeqFile.nextSequence()) {
                 final SAMSequenceRecord samSequenceRecord = makeSequenceRecord(refSeq);
+                //add aliases
+                if(contig2aliases.containsKey(samSequenceRecord.getSequenceName())) {
```

Comment:
To avoid hashing twice you could instead write
```
final Set<String> aliases = contig2aliases.get(samSequenceRecord.getSequenceName());
if (aliases != null) {
    ...
}
```